### DISCUSSION TITLE
GTF assembly with long reads / short reads for Isoform Switch detection

### DISCUSSION THREAD

**Discussion by Pinolinoo** [kvittingseerup/IsoformSwitchAnalyzeR#127] [General]
Dear Kristoffer, 

I am posting here a new question, regarding my analysis with your tool isoformswitch analyzer. I assembled my transcripts with the stringtie merge command and reanalyzed the bam files to possibly detect novel trancripts/ Isoformswitches. I assembled a new gtf file from short reads as well as from long reads coming from the same samples. 

After importRdata(), with fixStringTieAnnotationProblem = TRUE I get about 15626 annoated and 17408  unannoated transcripts. When I do the same with the gtf file resulting from the long reads I get 3953 unannoated and 32398 annoated transcripts. 

Is this something one would expect since the longer reads are more often uniquly mapped to already annoated transcripts whereas with short reads there is a greater uncertanty resulting in more un annoated sequences? 
Or may it be because the coverage is much deeper with the short reads than the long reads?

I appreciate your help, 
all the best 

philipp

**Comment by kvittingseerup** [comment]
Hi Philipp

Thanks for reaching out. Both of what you suggest are true:

- Short read will result in much higher false positives rates
- Long read have much lower coverage and hence will not identify as many transcripts

Have you run any QC tools ([gffcompare](https://ccb.jhu.edu/software/stringtie/gffcompare.shtml), [sqanti](https://github.com/ConesaLab/SQANTI), etc. ) on them?

If you want more info on this you need to post it on StringTie's github as it technically is a StringTie question ;-).

Btw Did you see that StringTie (v2.2.0+) can take *both* short and long reads as input?

Cheers
Kristoffer

**Reply by Pinolinoo** [reply]
Hi kristoffer, 
thanks for the reply! 

I will try to run the QC tools haven't tried that. The mixed mode of Strintie I am also currently trying out. Will let you know what comes out if you are interested. 

All the best 
Philipp

**Comment by Pinolinoo** [comment]
Ok, very interesting update! When using the mixed read mode from Stringtie for the GTF assambly and then correcting in isoformswitcheR() with with importRdata() and fixStringTieAnnotationProblem = TRUE  I get only 136 Stringtie IDs and 32690 annoated IDs!

**Comment by Pinolinoo** [comment]
I have another question: From the resulting Isoformswitcher analysis with the mixed merged gtf file I get now much more isoform switches in the range of up to 2700 thousand per comparison whereas before when I used only the short read assambled gtf I had around 300-600 per condition. Why is that the case?

**Reply by kvittingseerup** [reply]
Could I get you to post this as a new issue to make it easier to navigate/find things?

**Reply by Pinolinoo** [reply]
done!
